pkg/util/system/system_utils_test.go

```diff
@@ -21,27 +21,89 @@ import (
 
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/cmd/kubeadm/app/constants"
 )
 
 func TestIsMasterNode(t *testing.T) {
+	labelMater := map[string]string{
+		constants.LabelNodeRoleMaster: string(v1.TaintEffectNoSchedule),
+	}
 	testCases := []struct {
-		input  string
+		node   v1.Node
 		result bool
 	}{
-		{"foo-master", true},
-		{"foo-master-", false},
-		{"foo-master-a", false},
-		{"foo-master-ab", false},
-		{"foo-master-abc", true},
```

Comment:
Just curious, why `foo-master-abc` is a true case?
@jszczepkowski as you wrote the original code in #36219, could you please explain this? Thanks!

Reply:
Did anyone ever find out why you need `3` here? I have a case where my node is named `foo-master-1` (which seems reasonable) but I see in my logs that this has implications for HPA/metrics gathering (maybe among other things).  Trying to figure out the origins of the `3` @FengyunPan @jszczepkowski @xiangpengzhao